lib/app.js

```diff
@@ -3,10 +3,28 @@ const parseRequest = require('./utils/parseRequest');
 const createResponse = require('./utils/createResponse');
 
 const app = net.createServer(socket => {
-  socket.on('data', data => {
-    const request = parseRequest(data.toString());
-    socket.end(createResponse({ body: 'Not Found', status: '404 Not Found', contentType: 'text/plain' }));
-  });
+socket.on('data', data => {
+const request = parseRequest(data.toString());
+if (request.method === 'GET' && request.path === '/') {
+  socket.write(createResponse({ body: 'hi', contentType: 'text/plain', status: '200 OK'}))
+  return request.body;
+} else if (request.method === 'POST' && request.path === '/echo') {
+  socket.write(createResponse({body: 'hi', contentType: 'text/plain', status: '200 OK'}))
+} else if (request.method === 'GET' && request.path === '/red') {
+  socket.write(createResponse({ body: '<h1>red</h1>', contentType:'text/html', status: '200 OK' }))
+} else if (request.method === 'GET' && request.path === '/green') {
+  socket.write(createResponse({ body: '<h1>green</h1>', contentType:'text/html', status: '200 OK' }))
+} else if (request.method === 'GET' && request.path === '/blue') {
+  socket.write(createResponse({ body: '<h1>blue</h1>', contentType:'text/html', status: '200 OK' }))
+} else {
+socket.write(`HTTP/1.1 404 NOT FOUND\r
+Content-Type: text/plain\r
+\r
+Not found`)
+}
+
+socket.end(createResponse({ body: 'Not Found', status: '404 Not Found', contentType: 'text/plain' }));
```

Comment:
Because you're already sending a 'Not Found' message above, you can convert this to just close the connection without sending an additional response:

```suggestion
socket.end();
```